components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SubscriptionsApiServiceImpl.java

```diff
@@ -110,18 +111,45 @@ public Response subscriptionsGet(String apiId, String applicationId, String grou
 
                 // This will fail with an authorization failed exception if user does not have permission to access the API
                 ApiTypeWrapper apiTypeWrapper = apiConsumer.getAPIorAPIProductByUUID(apiId, organization);
-
+                String apiVersion;
+                String apiName;
                 if (apiTypeWrapper.isAPIProduct()) {
                     subscriptions = apiConsumer.getSubscribedIdentifiers(subscriber,
                             apiTypeWrapper.getApiProduct().getId(), groupId, organization);
+                    apiVersion = apiTypeWrapper.getApi().getId().getVersion();
+                    apiName = apiTypeWrapper.getApiProduct().getId().getName();
                 } else {
                     subscriptions = apiConsumer.getSubscribedIdentifiers(subscriber,
                             apiTypeWrapper.getApi().getId(), groupId, organization);
+                    apiVersion = apiTypeWrapper.getApi().getId().getVersion();
+                    apiName = apiTypeWrapper.getApi().getId().getApiName();
+                }
+                if (includeFromVersionRange) {
+                    String versionRange;
+                    try {
+                        versionRange = apiConsumer.getSubscriptionVersionRange(apiVersion, apiTypeWrapper.getUuid());
+                    } catch (APIManagementException e) {
+                        RestApiUtil.handleBadRequest("Version range is not accepted for the API", log);
+                        return null;
+                    }
```

Comment:
Do we need a try catch here? Just throw the error from the method itself. It will be handled from the REST API level.

Reply:
public Response subscriptionsGet(String apiId, String applicationId, String groupId, String xWSO2Tenant,
                                     Boolean includeFromVersionRange, Integer offset, Integer limit,
                                     String ifNoneMatch, MessageContext messageContext) throws APIManagementException